### PR TITLE
Lazy Plans

### PR DESCRIPTION
**Pull request by alxrsngrtn**
Fix for b/164161757 -- this PR introduces `properties` to `kotlin-generation-utils`. It also updates `plan-generator` to use lazy delegates for all global properties in generated recipe2plan files. 

Recommendation to the reviewer: View the diff without whitespace changes.

### CODE REVIEW

**Comment by alxrsngrtn** [comment]
CC: @jasonwyatt @jblebrun @piotrswigon

File: src/tools/tests/goldens/WriterReaderExample.kt [hunk 1]
```diff
@@ -15,258 +15,211 @@ import arcs.core.data.Plan.*
...
+val EphemeralWriting_Handle0 by lazy {
+    Handle(
+        StorageKeyParser.parse("create://my-ephemeral-handle-id"),
+        arcs.core.data.SingletonType(
             arcs.core.data.EntityType(
                 arcs.core.data.Schema(
                     setOf(arcs.core.data.SchemaName("Thing")),
                     arcs.core.data.SchemaFields(
-                        singletons = mapOf("name" to arcs.core.data.FieldType.Text),
+                        singletons = emptyMap(),
```
Comment (alxrsngrtn) [inline src/tools/tests/goldens/WriterReaderExample.kt:179]: Not sure why this changed, I'm going to investigate.

Comment (alxrsngrtn) [reply src/tools/tests/goldens/WriterReaderExample.kt:179]: Oh, this is a mixed up diff, nvm.

File: src/tools/kotlin-generation-utils.ts [hunk 2]
```diff
@@ -70,6 +70,26 @@ export class KotlinGenerationUtils {
     return this.applyFun('setOf', args, {startIndent, emptyName: 'emptySet'});
   }
 
+  async property(name: string, block: ({startIndent}) => Promise<string>, {
```
Comment (mmandlis) [inline src/tools/kotlin-generation-utils.ts:73]: plaese, add a comment, documenting this method?

**Comment by alxrsngrtn** [comment]
Presubmit check: cl/326587903

**Comment by alxrsngrtn** [comment]
I'm having weird merge issues w/ the presubmit check (likely due to resets and forced pushes). I'm going to postpone merging until I can get this sorted out.

**Comment by alxrsngrtn** [comment]
Also, see, cl/326769732

**Pull request merged**